src/main/scala/com/campudus/tableaux/TableauxRouter.scala

```diff
@@ -44,7 +45,16 @@ class TableauxRouter(verticle: Starter) extends Router with VertxAccess {
         }
       } yield x
     }
-    case Post(tableIdRows(tableId)) => getAsyncReply(controller.createRow(tableId.toLong))
+    case Post(tableIdRows(tableId)) => getAsyncReply {
+      val p = Promise[DomainObject]
+      getJson(req).onComplete {
+        case Success(json) =>
+          import scala.collection.JavaConverters._
+          controller.createFullRow(tableId.toLong, json.getArray("values").asScala.toSeq) map { p.success(_) } recover { case ex => p.failure(ex) }
+        case Failure(_) => controller.createRow(tableId.toLong) map { p.success(_) } recover { case ex => p.failure(ex) }
+      }
+      p.future
+    }

```

Comment:
Ok I want to use the same REST API call for empty/full row, because its the same. (full row of nothing is a empty row or a Json.obj()), so its make sense for me.
U will get a Future.failed with a NotFoundJsonException (but NoJsonFoundException is a better name), because other REST API calls need a JSON.

Now i found a better solution with flatMap and recoverWith, found my fail i did it before.
From where should the Option comes?
Before I had ColumnIds also in the JSON, but after I thought about it I refactored it, but now i want it back.
But which one:
Json.obj("columnIds" -> Json.arr(1, 2, 3), "values" -> Json.arr("test", 1, "something"))
Json.obj("1" -> "test", "2" -> 1, "3" -> "something")